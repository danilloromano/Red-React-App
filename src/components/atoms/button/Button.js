import React, { Component } from 'react';
import './Button.scss';

class Button extends Component {

    render() {
        return (
        <div className="btn__wrapper">
            <button className="btn__default" onClick={this.props.handleClick}>
                {this.props.text}
            </button>
        </div>
        )
    }
}

export default Button;
