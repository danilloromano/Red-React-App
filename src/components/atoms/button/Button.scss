@import "../../../assets/variables/variables.scss";

.btn {
    &__wrapper {
        padding-top: 60px;
    }

    &__default {
        width: 280px;
        height: 50px;
        background-color: $red;
        border-radius: 10px;
        cursor: pointer;
        font-size: 17px;
        text-align: center;
        color: $white;
        border: none;
    }
}




