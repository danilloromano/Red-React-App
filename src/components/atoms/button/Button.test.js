import React from 'react';
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme';
import Button from './Button';

describe('Button render test', () => {
   const handleClick = jest.fn();
   
    const defaultProps = {
        handleClick,
        text: 'OI'
    }

    defaultProps.handleClick.mockClear();

    it('should render correctly with props', () => {
        const wrapper = shallow(<Button {...defaultProps} />)
        expect(toJson(wrapper)).toMatchSnapshot()
    });

    it('should click on button and call handleClick funcion props', () => {
        const wrapper = shallow(<Button {...defaultProps} />)
        wrapper.find('.btn__default').simulate('click');
        expect(defaultProps.handleClick).toHaveBeenCalled();
    });
});
