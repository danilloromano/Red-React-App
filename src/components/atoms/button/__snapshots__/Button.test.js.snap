// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button render test should render correctly with props 1`] = `
<div
  className="btn__wrapper"
>
  <button
    className="btn__default"
    onClick={[MockFunction]}
  >
    OI
  </button>
</div>
`;
